mapbox_tilesets/scripts/cli.py

```diff
@@ -513,19 +516,42 @@ def add_source(ctx, username, id, features, no_validation, token=None, indent=No
         for feature in features:
             if not no_validation:
                 utils.validate_geojson(feature)
+
             file.write((json.dumps(feature) + "\n").encode("utf-8"))
 
         file.seek(0)
+
         m = MultipartEncoder(fields={"file": ("file", file)})
-        resp = s.post(
-            url,
-            data=m,
-            headers={
-                "Content-Disposition": "multipart/form-data",
-                "Content-type": m.content_type,
-            },
+        prog = click.progressbar(
+            length=m.len, fill_char="=", width=0, label="upload progress"
         )
 
+        if quiet:
```

Comment:
I'm sure there's a cleaner way to write this conditional, but I kept it verbose since the `callback` is required when defining the `MultipartEncoderMonitor`, but also can only be defined after `with prog:`